src/main/java/com/minecolonies/api/tools/ModToolTypes.java

```diff
@@ -0,0 +1,242 @@
+import com.minecolonies.api.util.constant.translation.ToolTranslationConstants;
+import io.netty.util.Constant;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.item.*;
+import net.minecraftforge.common.ToolAction;
+import net.minecraftforge.common.ToolActions;
+import net.minecraftforge.eventbus.api.IEventBus;
+import net.minecraftforge.registries.DeferredRegister;
+import net.minecraftforge.registries.RegistryObject;
+
+import java.util.Set;
+import java.util.function.Consumer;
+
+/**
+ * Class used for storing and registering any ToolTypes.
+ */
+public class ModToolTypes
+{
+    public final static DeferredRegister<ToolTypeEntry> DEFERRED_REGISTER = DeferredRegister.create(new ResourceLocation(Constants.MOD_ID, "tooltypes"), Constants.MOD_ID);
+
+    public static final String NONE_ID            = "none";
+    public static final String PICKAXE_ID         = "pickaxe";
+    public static final String SHOVEL_ID          = "shovel";
+    public static final String AXE_ID             = "axe";
+    public static final String HOE_ID             = "hoe";
+    public static final String SWORD_ID           = "sword";
+    public static final String BOW_ID             = "bow";
+    public static final String FISHING_ROD_ID     = "rod";
+    public static final String SHEARS_ID          = "shears";
+    public static final String SHIELD_ID          = "shield";
+    public static final String HELMET_ID          = "helmet";
+    public static final String LEGGINGS_ID        = "leggings";
+    public static final String CHESTPLATE_ID      = "chestplate";
+    public static final String BOOTS_ID           = "boots";
+    public static final String FLINT_AND_STEEL_ID = "flintandsteel";
+
+    public static RegistryObject<ToolTypeEntry> none;
+    public static RegistryObject<ToolTypeEntry> pickaxe;
+    public static RegistryObject<ToolTypeEntry> shovel;
+    public static RegistryObject<ToolTypeEntry> axe;
+    public static RegistryObject<ToolTypeEntry> hoe;
+    public static RegistryObject<ToolTypeEntry> sword;
+    public static RegistryObject<ToolTypeEntry> bow;
+    public static RegistryObject<ToolTypeEntry> fishing_rod;
+    public static RegistryObject<ToolTypeEntry> shears;
+    public static RegistryObject<ToolTypeEntry> shield;
+    public static RegistryObject<ToolTypeEntry> helmet;
+    public static RegistryObject<ToolTypeEntry> leggings;
+    public static RegistryObject<ToolTypeEntry> chestplate;
+    public static RegistryObject<ToolTypeEntry> boots;
+    public static RegistryObject<ToolTypeEntry> flint_and_steel;
+
+    static
+    {
+        none = register(NONE_ID,
+          builder -> builder.setDisplayName(Component.translatable(ToolTranslationConstants.TOOL_TYPE_NONE))
+                  .setIsTool(itemStack -> true)
+                  .setToolLevel(itemStack -> itemStack == null ? 0 : -1)
```

Comment:
As said in the previous comment, this should always return -1, not conditionally. Even if the itemstack is null or not.

Reply:
Done.